spyder/plugins/editor/panels/scrollflag.py

```diff
@@ -82,6 +90,37 @@ def update_flag_colors(self, color_dict):
             self._facecolors[name] = QColor(color)
             self._edgecolors[name] = self._facecolors[name].darker(120)
 
+    def delayed_update_flags(self):
+        """Call update flags later."""
+        if self._update_list_timer:
+            return
+
+        self._update_list_timer = QTimer(self)
+        self._update_list_timer.setSingleShot(True)
+        self._update_list_timer.timeout.connect(self.update_flags)
+        self._update_list_timer.start(REFRESH_RATE)
+
+    def update_flags(self):
+        """Update flags list."""
+        self._update_list_timer = None
+        self._todo_list = []
+        self._code_analysis_list = []
+        self._breakpoint_list = []
+        editor = self.editor
+        block = editor.document().firstBlock()
+        while block.isValid():
+            data = block.userData()
+            if data:
+                if data.code_analysis:
+                    self._code_analysis_list.append((block, data))
+                if data.todo:
+                    self._todo_list.append((block, data))
+                if data.breakpoint:
+                    self._breakpoint_list.append((block, data))
+            block = block.next()
```

Comment:
for readability (since the if are not if-elif....) I suggest to change to

```python

                if data.code_analysis:
                    self._code_analysis_list.append((block, data))

                if data.todo:
                    self._todo_list.append((block, data))

                if data.breakpoint:
                    self._breakpoint_list.append((block, data))
```

Reply:
Done